src/bidiMapper/domains/context/BrowsingContextProcessor.ts

```diff
@@ -434,19 +425,17 @@ export class BrowsingContextProcessor {
   #handleWorkerTarget(cdpTarget: CdpTarget, browsingContextId: string) {
     cdpTarget.cdpClient.on('Runtime.executionContextCreated', (params) => {
       const {uniqueId, id, origin} = params.context;
-      const realm = new Realm(
-        this.#realmStorage,
-        this.#browsingContextStorage,
-        uniqueId,
-        browsingContextId,
-        id,
-        serializeOrigin(origin),
-        'dedicated-worker',
-        undefined,
+      const realm = new DedicatedWorkerRealm(
         cdpTarget.cdpClient,
         this.#eventManager,
-        this.#sharedIdWithFrame,
-        this.#logger
+        id,
+        this.#logger,
+        serializeOrigin(origin),
+        this.#realmStorage.getRealm({
+          browsingContextId,
+        }),
```

Comment:
You probably want to have window realm here:
```suggestion
        this.#realmStorage.getRealm({
          browsingContextId,
          type: 'window'
        }),
```

Reply:
Only window realms carry a browsing context directly, so this is clear by definition.

Reply:
https://github.com/GoogleChromeLabs/chromium-bidi/pull/1709/files#r1453401191